osquery/tables/events/linux/containerd_events.cpp

```diff
@@ -0,0 +1,333 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/flags.h>
+#include <osquery/logger/logger.h>
+#include <osquery/registry/registry_factory.h>
+#include <osquery/sql/sql.h>
+
+#include <boost/algorithm/string/join.hpp>
+
+#include <osquery/events/eventpublisher.h>
+#include <osquery/events/eventsubscriber.h>
+#include <osquery/events/linux/containereventpublisher.h>
+
+namespace osquery {
+
+namespace {
+std::string labelsMapToString(
+    std::unordered_map<std::string, std::string> labels) {
+  std::string labels_string;
+  for (const auto& label_pair : labels) {
+    labels_string += label_pair.first + "=" + label_pair.second + ",";
+  }
+
+  if (!labels_string.empty()) {
+    labels_string.pop_back();
+  }
+  return labels_string;
+}
+} // namespace
+
+class ContainerdContainerEventSubscriber final
+    : public EventSubscriber<ContainerEventPublisher> {
+ public:
+  Status init() override;
+
+  Status Callback(const ECRef& ec, const SCRef& sc);
+};
+
+REGISTER(ContainerdContainerEventSubscriber,
+         "event_subscriber",
+         "containerd_container_events");
+
+Status ContainerdContainerEventSubscriber::init() {
+  auto subscription = createSubscriptionContext();
+  subscription->event_type_subscription =
+      BaseContainerEvent::Type::ContainerEvent;
+
+  subscribe(&ContainerdContainerEventSubscriber::Callback, subscription);
+  return Status::success();
+}
+
+Status ContainerdContainerEventSubscriber::Callback(const ECRef& ec,
+                                                    const SCRef& sc) {
+  if (ec->container_event->type_ != BaseContainerEvent::Type::ContainerEvent) {
+    return Status::failure(
+        "Received wrong event type: " +
```

Comment:
how frequently do we get wrong event type here? This is probably just defensive coding? Do you think it makes sense to log the error here before returning?